Dispatcher/UI/Profile/Settings/SettingsViewModel.swift

```diff
@@ -0,0 +1,49 @@
+import Foundation
+
+class SettingsViewModel {
+    
+    let repository = SettingsRepository()
+    
+    var appSettings: [SettingModel] = [
+        SettingModel(sectionTitle: "Search results", options: [
+            SingleSetting(title: "Save filters", description: "Allow us to save filters when entering back to the app", status: .off),
+            SingleSetting(title: "Save search results", description: "Allow us to save your search result preferences for next search", status: .off)
+        ]),
+        SettingModel(sectionTitle: "App preferences", options: [
+            SingleSetting(title: "Notification", status: .on)
+        ]),
+    ]
+    
+    func fetchUserSettingsPreferences() {
+        appSettings[0].options[0].status = repository.fetchSaveFiltersSetting()
+        appSettings[0].options[1].status = repository.fetchSaveSearchResultsSetting()
+        appSettings[1].options[0].status = repository.fetchNotificationsSetting()
+    }
+    
+    func updateSetting(settingTitle: String, completionHandler: @escaping () -> ()) {
+        var modelIndex = -1
+        var settingIndex = -1
+        for (i, settingModel) in appSettings.enumerated() {
```

Comment:
We don't want to get situation of O(n^2) - loop inside loop. this is bad for runtime performance.
Usually we can fix it using different data structure as Dictionary (hashmap).